frontend/apps/app/app/api/webhook/github/utils/checkSchemaChanges.ts

```diff
@@ -0,0 +1,61 @@
+import { getPullRequestFiles } from '@/libs/github/api.server'
+import { savePullRequestTask } from '@/src/trigger/jobs'
+import { prisma } from '@liam-hq/db'
+import { minimatch } from 'minimatch'
+
+type CheckSchemaChangesParams = {
+  pullRequestNumber: number
+  pullRequestTitle: string
+  projectId: number
+  owner: string
+  name: string
+  repositoryId: number
+}
+
+export const checkSchemaChanges = async (
+  params: CheckSchemaChangesParams,
+): Promise<{ shouldContinue: boolean }> => {
+  const { pullRequestNumber, projectId, owner, name, repositoryId } = params
+
+  // Get changed files from pull request
+  const files = await getPullRequestFiles(
+    repositoryId,
+    owner,
+    name,
+    pullRequestNumber,
+  )
+  const filenames = files.map((file) => file.filename)
+
+  // Get project information
+  const project = await prisma.project.findUnique({
+    where: { id: projectId },
+    include: { watchSchemaFilePatterns: true },
+  })
+
+  if (!project || project.watchSchemaFilePatterns.length === 0) {
+    return { shouldContinue: false }
+  }
+
+  // Check if filenames match the patterns
+  const matchedFiles = filenames.filter((filename) =>
+    project.watchSchemaFilePatterns.some((pattern) =>
+      minimatch(filename, pattern.pattern),
+    ),
+  )
```

Comment:
The current implementation in `checkSchemaChanges.ts` fetches the entire project with its associated `watchSchemaFilePatterns` using `prisma.project.findUnique()`. We can optimize this query by directly searching in the `watchSchemaFilePatterns` table since we only need the patterns for a given project.

```suggestion
  // Get patterns for the project
  const patterns = await prisma.watchSchemaFilePattern.findMany({
    where: { projectId },
    select: { pattern: true }
  })

  if (patterns.length === 0) {
    return { shouldContinue: false }
  }

  // Check if filenames match the patterns
  const matchedFiles = filenames.filter((filename) =>
    patterns.some((pattern) =>
      minimatch(filename, pattern.pattern),
    ),
  )
```

This approach is more efficient because:
1. It only fetches the required data (patterns) rather than the entire project
2. It reduces the amount of data transferred from the database
3. The query is more straightforward and easier to understand

Reply:
I now understand that as the project grows larger, unnecessary data retrieval becomes a bigger issue.
Based on this, I have applied the fix in the following commit and confirmed that all tests pass successfully.

I will merge it once the CI passes!
[🔧 refactor: optimize Prisma query in checkSchemaChanges](https://github.com/liam-hq/liam/pull/918/commits/eb77e739177a5fd528291e184453dfb27de505b1)

Reply:
Thanks for adjusting!!